deploy/ray-operator/controllers/common/pod.go

```diff
@@ -0,0 +1,129 @@
+package common
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	rayiov1alpha1 "ray-operator/api/v1alpha1"
+	"strconv"
+	"strings"
+)
+
+type PodConfig struct {
+	RayCluster  *rayiov1alpha1.RayCluster
+	PodTypeName string
+	PodName     string
+	Extension   rayiov1alpha1.Extension
+}
+
+func DefaultPodConfig(instance *rayiov1alpha1.RayCluster, podTypeName string, podName string) *PodConfig {
+	return &PodConfig{
+		RayCluster:  instance,
+		PodTypeName: podTypeName,
+		PodName:     podName,
+	}
+}
+
+// build a pod for the cluster instance.
+func BuildPod(conf *PodConfig) *corev1.Pod {
+	// build label for cluster
+	rayLabels := labelsForCluster(*conf.RayCluster, conf.PodName, conf.PodTypeName, conf.Extension.Labels)
+
+	// build container for pod, now only handle one container for each pod
+	var containers []corev1.Container
+	container := buildContainer(conf)
+	containers = append(containers, container)
+
+	// create volume
+	volumes := conf.Extension.Volumes
+
+	spec := corev1.PodSpec{
+		Volumes:            volumes,
+		Containers:         containers,
+		Affinity:           conf.Extension.Affinity,
+		Tolerations:        conf.Extension.Tolerations,
+		ServiceAccountName: conf.RayCluster.Namespace,
+	}
+
+	// build annotations and store podCompareHash for comparison
+	annotations := conf.Extension.Annotations
+
+	pod := &corev1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "Pod",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        conf.PodName,
+			Namespace:   conf.RayCluster.Namespace,
+			Labels:      rayLabels,
+			Annotations: annotations,
+		},
+		Spec: spec,
+	}
+
+	return pod
+}
+
+func buildContainer(conf *PodConfig) corev1.Container {
+
+	redisPort, _ := strconv.ParseInt(defaultRedisPort, 10, 64)
```

Comment:
Why we define these ports as string and parse them to int here? Can we use int32 directly in constant.go?

Reply:
Good advice~